src/libsyntax/parse/parser.rs

```diff
@@ -233,6 +233,21 @@ macro_rules! maybe_whole {
     )
 }
 
+/// Uses $parse_expr to parse an expression and returns the span of the interpolated
+/// token or the span of the parsed expression, if it was not interpolated
+macro_rules! interpolated_or_expr_span {
```

Comment:
It's a macro so it can be used with different methods for parsing expressions. In particular it is once used with [Parser::parse_bottom_expr](https://github.com/rust-lang/rust/pull/31089/files#diff-da9d34ca1d0f6beee2838cf02e07345cR2342) and also with [Parser::parse_prefix_expr](https://github.com/rust-lang/rust/pull/31089/files#diff-da9d34ca1d0f6beee2838cf02e07345cR2751).

I've update the macro to return `Result`, which should have the same effect making it a `fn`, at least with respect to the `try!` (I hope). A similar `fn` would have to take a closure which is probably slightly less ergomatic. However if you still think a `fn` would be better, I'd be happy to change that.